GeoLlama/calc_by_slice.py

```diff
@@ -280,41 +269,43 @@ def evaluate_slice(slice_coords: list,
     mask_s1 = np.argwhere(view**2 >= thres)
 
     # Skip slice if no pixels masked
-    # if len(mask_s1) < pt_thres:
-    #     return
-    # centroid_s1 = 0.5*(mask_s1.max(axis=0) + mask_s1.min(axis=0))
     centroid_s1 = mask_s1.mean(axis=0)
 
     # Step 2: Use Mahalanobis distance to remove potential outliers
-    covar_inv = np.linalg.inv(np.cov(mask_s1.T))
-    pts_deviation = mask_s1 - centroid_s1
-    maha_dist = np.dot(np.dot(pts_deviation, covar_inv), pts_deviation.T).diagonal()
-    p_val = 1 - chi2.cdf(np.sqrt(maha_dist), 1)
-    mask_s2 = np.delete(mask_s1, np.argwhere(p_val<0.07), axis=0)
+    jackknife_dist = np.empty(len(mask_s1))
+    for idx in range(len(mask_s1)):
+        temp = np.delete(mask_s1, idx, axis=0)
+        S_inv = np.linalg.inv(np.cov(temp.T))
+        diffs = (mask_s1[idx] - temp.mean(axis=0))[np.newaxis, :]
+        jackknife_dist[idx] = np.linalg.norm( (diffs @ S_inv.T @ diffs.T).diagonal() )
 
-    # Skip slice if no pixels masked
-    # if len(mask_s2) < pt_thres:
-    #     return
+    conf_limit = t.interval(confidence=0.999, df=2,
+                            loc=jackknife_dist.mean(), scale=sem(jackknife_dist))[1]
+    mask_s2 = np.squeeze(mask_s1[np.argwhere(jackknife_dist<=conf_limit)], axis=1)
 
     # Step 3: Use distance-based clustering to find sample region
-    clusters = DBSCAN(eps=15, min_samples=15).fit_predict(mask_s2)
+    clusters = DBSCAN(eps=15, min_samples=350).fit_predict(mask_s2)
     mask_s3_args = np.argwhere(clusters==mode(clusters, keepdims=True).mode)
     mask_s3 = mask_s2[mask_s3_args.flatten()]
 
     # Skip slice if no pixels masked
```

Comment:
do the lines below actually skip the slice? I'm a little confused by this comment